core/src/main/java/org/infinispan/util/concurrent/locks/PendingLockPromise.java

```diff
@@ -0,0 +1,64 @@
+package org.infinispan.util.concurrent.locks;
+
+/**
+ * A promise returned by {@link PendingLockManager}.
+ * <p>
+ * When a transaction need to waits for older topology transaction, this class allows it check the state. If the
+ * transaction does not need to wait, or all older transaction has finished or has timed out, the {@link #isReady()}
+ * method returns {@code true}. Also, it allows to add listeners to be notified when it is ready.
```

Comment:
Some grammar issues here: needs to wait, allows it to check, all older transactions have finished or have timed out, it allows adding (or it allows the caller to add).